src/core/server/saved_objects/migrations/actions/create_index.ts

```diff
@@ -140,19 +141,25 @@ export const createIndex = ({
   return pipe(
     createIndexTask,
     TaskEither.chain<
-      RetryableEsClientError | IndexNotYellowTimeout | ClusterShardLimitExceeded,
+      RetryableEsClientError | IndexNotGreenTimeout | ClusterShardLimitExceeded,
       AcknowledgeResponse,
       'create_index_succeeded'
     >((res) => {
       if (res.acknowledged && res.shardsAcknowledged) {
-        // If the cluster state was updated and all shards ackd we're done
+        // If the cluster state was updated and all shards started we're done
         return TaskEither.right('create_index_succeeded');
       } else {
-        // Otherwise, wait until the target index has a 'yellow' status.
+        // Otherwise, wait until the target index has a 'green' status meaning
+        // the primary (and on multi node clusters) the replica has been started
         return pipe(
```

Comment:
Let's see if I got this right:

Thanks to the `timeout: 29s` we now allow ES to respond with a _200 OK_ before elasticsearch-js's 30s _requestTimeout_.

At this point, `acknowledged` and/or `shardsAcknowledged` might be false, which will cause the flow to execute `waitForIndexStatus`,  effectively waiting for all shards to create the index (polling using cluster.health API).

We previously did not enter this branch of the code, because `JS client timeout < ES timeout`, which caused the logic to systematically retry the whole step.

For the sake of review, I just made sure that in case of ES timeout, the response is still `200 OK`, otherwise we would not enter here. I checked this in Console, creating a dummy index with a 2 nanoseconds timeout:

![image](https://user-images.githubusercontent.com/25349407/180963265-e8e2d6b5-9789-4396-aca5-e7812d648221.png)

Reply:
I couldn't reproduce this so I asked the ES team to confirm the behaviour, but I didn't realise I could use less than second resolution for the timeouts. I've added an integration test with `timeout=2nanos` 😄